apps/ejabberd/src/node_push.erl

```diff
@@ -97,26 +98,24 @@ publish_item(Nidx, Publisher, Model, MaxItems, ItemId, ItemPublisher, Payload, P
 
     case is_allowed_to_publish(Model, Affiliation) of
         true ->
-            do_publish_item(Nidx, Publisher, Model, MaxItems, ItemId, ItemPublisher,
-                            Payload, PublishOptions);
+            do_publish_item(ServerHost, PublishOptions, Payload);
         false ->
             {error, ?ERR_FORBIDDEN}
     end.
 
-do_publish_item(_Nidx, _Publisher, _Model, _MaxItems, _ItemId, _ItemPublisher,
-                [#xmlel{name = <<"notification">>} | _] = Notifications, PublishOptions) ->
-    Host = ?MYNAME,
+do_publish_item(ServerHost, PublishOptions,
+                [#xmlel{name = <<"notification">>} | _] = Notifications) ->
```

Comment:
Is it possible the `#xmlel{name = <<"notification">>}` will not be the first element?